### DISCUSSION TITLE
NodeJS-LibP2P: Listening on WebRTC-Direct MultiAddr

### DISCUSSION THREAD

**Discussion by AustinFoss** [libp2p/js-libp2p#2038] [Q&A]
Previously asked on the LibP2P forums: https://us06web.zoom.us/j/8280739255?pwd=d3RzRXUrakhiUHFTS2J0VHB3TU4rUT09

Hello,

I’ve spent a few days, off and on trying to figure out why I can’t seem to get my node.js-libp2p node to advertise a webrtc-direct listening multiaddr. Recently had a discussion over in the IPFS discord with another new dev who pointed me to this [pull request](https://github.com/libp2p/js-libp2p/pull/1905).

At the top, the pull request's OP states that “Listening private-to-public” is not supported, and I’m just kinda wondering why that might be? Is there something technically limiting about node.js that prevents this?

**Comment by maschad** [comment]
As mentioned [here](https://github.com/libp2p/js-libp2p/pull/1905#issuecomment-1710249382) there is currently a [draft PR](https://github.com/libp2p/js-libp2p/pull/2027) to add this, but the underlying library [would need to add an option to disable config validation first](https://github.com/paullouisageneau/libdatachannel/issues/970)
